src/coreclr/tools/Common/CommandLineHelpers.cs

```diff
@@ -69,22 +69,21 @@ public static TargetOS GetTargetOS(string token)
 
                 throw new NotImplementedException();
             }
-            else
+
+            string lowerToken = token.ToLowerInvariant();
```

Comment:
Maybe future proof this with something like the following?

```suggestion
            int dashIndex = token.IndexOf('-');
            if (dashIndex > 0)
            {
                token = token[..dashIndex];
            }
            string lowerToken = token.ToLowerInvariant();
```

Reply:
This would mean that the compiler is going to happily accept a conflicting command line like `--targetarch x64 --targetos win-x64` instead of reporting an error.

I do not think we need to future proof this. We can always relax it once we find a good reason to.